packages/driver/cypress/e2e/cypress/cy.cy.js

```diff
@@ -565,5 +565,54 @@ describe('driver/src/cypress/cy', () => {
 
       cy.get('body').find('#specific-contains').children().should('have.class', 'active')
     })
+
+    context('overwriting queries', () => {
+      it('does not allow commands to overwrite queries', () => {
+        const fn = () => Cypress.Commands.overwrite('get', () => {})
+
+        expect(fn).to.throw().with.property('message')
+        .and.include('Cannot overwite the `get` query. Queries can only be overwritten with `Cypress.Commands.overwriteQuery()`.')
+
+        expect(fn).to.throw().with.property('docsUrl')
+        .and.include('https://on.cypress.io/api')
+      })
+
+      it('does not allow queries to overwrite commands', () => {
+        const fn = () => Cypress.Commands.overwriteQuery('click', () => {})
+
+        expect(fn).to.throw().with.property('message')
+        .and.include('Cannot overwite the `click` command. Commands can only be overwritten with `Cypress.Commands.overwrite()`.')
+
+        expect(fn).to.throw().with.property('docsUrl')
+        .and.include('https://on.cypress.io/api')
+      })
+
```

Comment:
can we test to ensure the context is correctly passed though?

Reply:
Good idea, will add a test for that.